Summary: rpm: fix segment fault in psm.c

When rpmtsGetRdb(ts) retrun NULL, it will output segment fault.

Upstream-status: pending
    
Signed-off-by: Zhixiong Chi <zhixiong.chi@windriver.com>
Index: rpm-5.4.14/lib/psm.c
===================================================================
--- rpm-5.4.14.orig/lib/psm.c	2015-07-15 17:01:13.238839862 +0800
+++ rpm-5.4.14/lib/psm.c	2015-07-15 17:21:56.914828001 +0800
@@ -3026,13 +3026,15 @@
 	     rc = rpmpsmNext(psm, PSM_CHROOT_OUT);
 #endif
 
-	xx = rpmtxnBegin(rpmtsGetRdb(ts), _parent, NULL);
-	rc = runInstScript(psm);
-	if (rc)
-	     xx = rpmtxnAbort(rpmtsGetRdb(ts)->db_txn);
-	else
-	     xx = rpmtxnCommit(rpmtsGetRdb(ts)->db_txn);
-	rpmtsGetRdb(ts)->db_txn = NULL;
+	if (rpmtsGetRdb(ts)) {
+	    xx = rpmtxnBegin(rpmtsGetRdb(ts), _parent, NULL);
+	    rc = runInstScript(psm);
+	    if (rc)
+		xx = rpmtxnAbort(rpmtsGetRdb(ts)->db_txn);
+	    else
+		xx = rpmtxnCommit(rpmtsGetRdb(ts)->db_txn);
+	    rpmtsGetRdb(ts)->db_txn = NULL;
+	}
 #ifdef RPM_VENDOR_OE
 	if (scriptletWrapper && *scriptletWrapper)
 	     rc = rpmpsmNext(psm, PSM_CHROOT_IN);
